stan/math/prim/functor/algebra_solver_fp.hpp

```diff
@@ -0,0 +1,306 @@
+#ifndef STAN_MATH_PRIM_FUNCTOR_FP_SOLVER_HPP
+#define STAN_MATH_PRIM_FUNCTOR_FP_SOLVER_HPP
```

Comment:
Does the fixed point function have a language design thing? Asking cause the signature is different than the others.

Reply:
I'm not sure what the answer to this is—@charlesm93 indicated above that we should do the fixed point solver as well. Where might I look to find the language design?

Reply:
Oh okay, then the language stuff and documentation will need added (the other changes are just silently updating how `algebra_solver` and `algebra_solver_newton` already work), but that doesn't have to go with the Math PR.

Reply:
Ah, ok, cool—so, concretely, should I factor out all the changes to the fixed point solver into a separate PR, or should I leave it as is and open up a PR for language and documentation elsewhere?

Reply:
Oh woof, I just went and looked at this a little closer. Pull fixed point out into another PR. The signature is new and it isn't just a copy-paste (what I was assuming an hour ago). It will be easier to get Powell and Newton done this way, and won't be any harder to finish FP.

Since Charles is the one looking for the FP signatures, probably best to get him to review that anyway (I'm unfamiliar with this stuff).